zcash_client_backend/CHANGELOG.md

```diff
@@ -128,10 +128,14 @@ and this library adheres to Rust's notion of
 
 ### Removed
 - `zcash_client_backend::data_api`:
-  - `WalletRead::get_all_nullifiers`
-  - `WalletRead::{get_commitment_tree, get_witnesses}` have been removed
-    without replacement. The utility of these methods is now subsumed
-    by those available from the `WalletCommitmentTrees` trait.
+  - `WalletRead::block_height_extrema` has been removed. Use `chain_height`
```

Comment:
Why does this method need to be removed? It is still a valid query to ask "what are the minimum and maximum heights for blocks we have seen?" even if there is a separate new method for the wallet's view of the chain tip. The changelog entry seems to imply that this data now needs to be fetched directly from the wallet DB's SQLite tables, but we've been generally moving in the direction of having callers access data via Rust methods instead.

If this method were left in place, then `WalletRead::get_target_and_anchor_heights` could remain a provided trait method (because the minimum height would be accessible).

Reply:
My rationale for removing it is that while it's valid to ask what the maximum and minimum heights are that the wallet has seen, it's not really very meaningful because of all the gaps that may exist, and the minimum block height no longer corresponds to the wallet birthday; instead, it corresponds to the start of the shard containing the wallet birthday height. If we want to provide a single method, it should probably return a struct with `min_block_scanned`, `max_block_scanned`, `chain_tip`, and maybe `wallet_birthday` fields, and it should be named something different in order to ensure users switch away from the old method, which has muddled semantics.